apps/gitness/src/pages-v2/repo/transform-utils/__tests__/branch-transform.test.ts

```diff
@@ -0,0 +1,82 @@
+import { afterAll, beforeAll, describe, expect, it, vi } from 'vitest'
+
+import { ListBranchesOkResponse, TypesBranchExtended, TypesCommitDivergence } from '@harnessio/code-service-client'
+
+import { apiBranches2BranchNames, apiBranches2DefaultBranchName, transformBranchList } from '../branch-transform'
+
+// Mock data
+const mockBranches: TypesBranchExtended[] = [
+  {
+    name: 'main',
+    commit: { sha: 'abc123', committer: { when: '2025-01-02T00:00:00Z', identity: { name: 'John Doe' } } },
+    is_default: true
+  },
+  {
+    name: 'feature',
+    commit: { sha: 'def456', committer: { when: '2025-01-02T00:00:00Z', identity: { name: 'Jane Smith' } } },
+    is_default: false
+  }
+]
+const mockDivergence: TypesCommitDivergence[] = [
+  { ahead: 5, behind: 3 },
+  { ahead: 2, behind: 1 }
+]
+
+const mockApiBranches: ListBranchesOkResponse = [
+  { name: 'main', is_default: true },
+  { name: 'feature', is_default: false }
+]
+
+describe('transformBranchList', () => {
+  beforeAll(() => {
+    // Use fake timers to control the current time
+    vi.useFakeTimers()
+    // Set the system time to a specific date
+    vi.setSystemTime(new Date('2025-01-03T00:00:00Z'))
+  })
+
+  afterAll(() => {
+    // Restore real timers after tests
+    vi.useRealTimers()
+  })
```

Comment:
Faker timers controls things like `setTimeout` and `setInterval`, not system time. You can kill them off with these tests.
```suggestion
  beforeAll(() => {
    // Set the system time to a specific date
    vi.setSystemTime(new Date('2025-01-03T00:00:00Z'))
  })
```

Reply:
done